delivery_schenker/tests/test_delivery_schenker.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.tests import Form, common
+
+
+class TestDeliverySchenker(common.SavepointCase):
+    maxDiff = None
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.carrier = cls.env["delivery.carrier"].create(
+            {
+                "name": "Schenker",
+                "schenker_access_key": "key",
+                "schenker_group_id": "group",
+                "schenker_user": "user",
+                "schenker_barcode_format": "A4",
+                "schenker_barcode_mail": "test@test.com",
+                "schenker_booking_type": "land",
+                "schenker_incoterm_id": cls.env.ref("account.incoterm_EXW").id,
+                "schenker_service_type": "D2D",
+                "schenker_service_land": "CON",
+                "schenker_indoor_delivery": True,
+                "schenker_express": True,
+                "schenker_food_related": True,
+                "schenker_heated_transport": True,
+                "schenker_home_delivery": True,
+                "schenker_own_pickup": True,
+                "schenker_pharmaceuticals": True,
+                "schenker_default_packaging_id": cls.env.ref(
+                    "delivery_schenker.schenker_packaging_01"
+                ).id,
+                "schenker_address_number": "address_number",
+                "product_id": cls.env.ref("delivery.product_product_delivery").id,
+            }
+        )
+        cls.company = cls.env.company
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "Test partner",
+                "country_id": cls.company.partner_id.country_id.id,
+                "phone": cls.company.partner_id.phone,
+                "email": "test@odoo.com",
+                "street": cls.company.partner_id.street,
+                "city": cls.company.partner_id.city,
+                "zip": cls.company.partner_id.zip,
+                "state_id": cls.company.partner_id.state_id.id,
+                "vat": cls.company.partner_id.vat,
+            }
+        )
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test product",
+                "type": "product",
+                "weight": 1,
+                "volume": 1,
+            }
+        )
+        cls.sale, cls.picking = cls._create_sale_order(cls)
+        cls._process_picking(cls, cls.picking)
+
+    def _process_picking(self, picking):
+        picking.move_lines.quantity_done = 1
+        picking.button_validate()
+        # Delete result package, otherwise Unittests will fail
+        # because the addon base_delivery_carrier_label gets installed first
+        # which creates a default package within the method _set_a_default_package
+        picking.move_line_ids.result_package_id = False
```

Comment:
Yes i know, but i dont Like it to add Logic from an other module which is not in the dependency list. We only have to add this in the Tests because of the oca unitests behavior. If you Test it separate everything works fine.